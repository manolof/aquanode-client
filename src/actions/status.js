import { connect } from 'socket.io-client';

const socket = connect('/status');

export const StatusActions = Object.freeze({
	GET: '[Status] GET',
});

export const getStatusAction = () => (dispatch) => {
	socket.on('get', (message) => {
		dispatch({
			type: StatusActions.GET,
			payload: message.data,
		});
	});
};
