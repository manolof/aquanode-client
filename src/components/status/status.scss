@import '../../styles/shared.scss';

.container {
	display: flex;
	justify-content: space-between;
}
