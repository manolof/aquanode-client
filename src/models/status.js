import PropTypes from 'prop-types';

export const StatusModel = PropTypes.shape({
	time: PropTypes.string,
	entities: PropTypes.arrayOf(
		PropTypes.shape({
			type: PropTypes.string,
			status: PropTypes.string,
		}),
	),
});
