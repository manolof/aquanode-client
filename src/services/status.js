export const fetchStatus = () => {
	return fetch('/status')
		.then((res) => res.json())
		.then((res) => res.data);
};
