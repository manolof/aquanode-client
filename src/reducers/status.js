import { StatusActions } from '../actions/status';

export const initialState = {
	time: '',
	entities: [],
};

const statusReducer = (state = initialState, action) => {
	switch (action.type) {
		case StatusActions.GET:
			return {
				...state,
				...action.payload,
			};

		default:
			return state;
	}
};

export {
	statusReducer,
};
