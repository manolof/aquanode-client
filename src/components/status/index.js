import React, { Component } from 'react';

import { StatusModel } from '../../models/status';

class Status extends Component {
	render() {
		const { status } = this.props;

		return (
			<div className="Status">
				<h4>The date now:</h4>
				{
					<p>{status.time}</p>
				}

				<h4>Status:</h4>
				{
					status.entities.map((entity) =>
						<p key={entity.type}>{entity.type} : {entity.status}</p>,
					)
				}
			</div>
		);
	}
}

Status.propTypes = {
	status: StatusModel.isRequired,
};

export default Status;
