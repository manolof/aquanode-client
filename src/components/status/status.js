import React, { Component } from 'react';
import moment from 'moment';

import './status.css';
import { StatusModel } from '../../models/status';

class Status extends Component {
	render() {
		const { status } = this.props;

		return (
			<div className="container">
				<div className="server-time">
					<strong>Server time:&nbsp;</strong>
					{moment(status.time).format('MMM DD, HH:mm:ss')}
				</div>

				<div className="statuses">
					{
						status.entities.map((entity) =>
							entity.status &&
							<span key={entity.type}>
								{entity.type}: <strong>{entity.status}</strong>&nbsp;
							</span>,
						)
					}
				</div>
			</div>
		);
	}
}

Status.propTypes = {
	status: StatusModel.isRequired,
};

export default Status;
