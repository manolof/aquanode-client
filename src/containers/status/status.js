import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import { getStatusAction } from '../../actions/status';
import Status from '../../components/status/status';
import { StatusModel } from '../../models/status';

class StatusContainer extends Component {
	componentDidMount() {
		const { getStatus } = this.props;
		getStatus();
	}

	render() {
		const { status } = this.props;

		return (
			<Status status={status}/>
		);
	}
}

StatusContainer.propTypes = {
	status: StatusModel.isRequired,
	getStatus: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
	status: state.status,
});

const mapDispatchToProps = (dispatch) => ({
	getStatus: () => dispatch(getStatusAction()),
});

export { StatusContainer };

export default connect(mapStateToProps, mapDispatchToProps)(StatusContainer);
