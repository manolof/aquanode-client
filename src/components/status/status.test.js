import { shallow } from 'enzyme';
import React from 'react';

import { Status } from './status';

describe('Status Component', () => {
	afterEach(() => {
		jest.resetAllMocks();
	});

	afterAll(() => {
		jest.restoreAllMocks();
	});

	it('should render', () => {
		const mockStatusProps = {
			time: '2018-10-23',
			entities: [
				{
					type: 'lights',
					status: 'night',
				},
			],
		};

		const wrapper = shallow(
			<Status status={mockStatusProps}/>,
		);

		expect(wrapper.getElements()).toMatchSnapshot();
	});
});
