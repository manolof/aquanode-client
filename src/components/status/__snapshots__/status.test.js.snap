// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should render 1`] = `
Array [
  <div
    className="container"
>
    <div
        className="server-time"
    >
        <strong>
            Server time: 
        </strong>
        Oct 23, 00:00:00
    </div>
    <div
        className="statuses"
    >
        <span>
            lights
            : 
            <strong>
                night
            </strong>
             
        </span>
    </div>
</div>,
]
`;
